### PR TITLE
Feature: Autostart for DE following the freedesktop standard

### PR DESCRIPTION
**Pull request by infeo**
This PR adds an implementation of the AutoStart SPI: The `FreeDesktopAutostartService`.

The implementation is based on the [freedesktop autostart specificiation](https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ Linux-specific implemenations of the [integrations-api](https://github.com/crypt
 
 This project uses the following JVM properties:
 * `cryptomator.integrationsLinux.trayIconsDir` - specifies the directory from which svg images for the tray icon are loaded
```
[the remark this answers is not in the document]
Comment (infeo) [reply README.md:7]: @coderabbitai out of scope for this PR.

File: README.md [hunk 2]
```diff
@@ -5,4 +5,4 @@ Linux-specific implemenations of the [integrations-api](https://github.com/crypt
...
-
+* `cryptomator.integrationsLinux.autoStartCmd` - specifies the command used for starting Cryptomator
```
[the remark this answers is not in the document]
Comment (infeo) [reply README.md:8]: @coderabbitai out of PR scope

**Review by overheadhunter — APPROVED** [review]

File: src/main/java/org/cryptomator/linux/autostart/FreedesktopAutoStartService.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		this.content = CONTENT_TEMPLATE.formatted(execValue, this.getClass().getName());
+	}
+
+	@Override
+	public synchronized void enable() throws ToggleAutoStartFailedException {
+		try {
+			Files.writeString(autostartFile, content, StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+		} catch (IOException e) {
+			throw new ToggleAutoStartFailedException("Failed to activate Cryptomator autostart for GNOME desktop environment.", e);
+		}
```
[the remark this answers is not in the document]
Comment (infeo) [reply src/main/java/org/cryptomator/linux/autostart/FreedesktopAutoStartService.java:62]: @coderabbitai No. If the method fails, the exception is thrown directly to the caller containing all debugging information needed.

File: src/main/java/org/cryptomator/linux/autostart/FreedesktopAutoStartService.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	@Override
+	public synchronized void disable() throws ToggleAutoStartFailedException {
+		try {
+			Files.deleteIfExists(autostartFile);
+		} catch (IOException e) {
+			throw new ToggleAutoStartFailedException("Failed to deactivate Cryptomator autostart for GNOME desktop environment.", e);
+		}
```
[the remark this answers is not in the document]
Comment (infeo) [reply src/main/java/org/cryptomator/linux/autostart/FreedesktopAutoStartService.java:71]: @coderabbitai No. If the method fails, the exception is thrown directly to the caller containing all debugging information needed.

**Comment by infeo** [comment]
@overheadhunter Fyi, i synchronized the methods `enable`, `disable` and `isEnabled` to prevent racing conditions when editing the autostart file (e.g. a lib consumers calls furiosly these method). Also, since the isSuppported check is non-static, i added a condition when the `exec` field is empty.

**Pull request merged**